lib/Onyx.js

```diff
@@ -367,7 +371,14 @@ function disconnect(connectionID, keyToRemoveFromEvictionBlocklist) {
         removeFromEvictionBlockList(keyToRemoveFromEvictionBlocklist, connectionID);
     }
 
+    const key = callbackToStateMapping[connectionID].key;
     delete callbackToStateMapping[connectionID];
+
+    // When the last subscriber disconnects, drop cache as well
+    const remainingConnectionsForKey = _.filter(callbackToStateMapping, {key});
+    if (remainingConnectionsForKey.length === 0) {
+        cache.remove(key);
```

Comment:
Separately, doesn't this sort of invalidate at least some of the usefulness of having the cache. It seems like we are only optimizing for the case of multiple subscribers.

e.g. if it takes some non trivial amount of time to retrieve something like a large report from `AsyncStorage` if we remove it the cache once we are no longer subscribed that means we will have to incur the same cost again when connecting again.

Another option would be to clear the cache key based on some other criteria like no subscribers and also least recently subscribed. But not sure if we need this optimization at all. What is the worst that can happen if the cache entry is not removed at all and the cache is allowed to grow to the size of storage?